source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp

```diff
@@ -0,0 +1,1225 @@
+        if (uniformLocation >= 0)
+        {
+            ProgramInputPtr inputPtr = std::make_shared<ProgramInput>(uniformLocation, uniformType, uniformSize);
+            _uniformList[std::string(uniformName)] = inputPtr;
+        }
+    }
+    delete[] uniformName;
+
+    return _uniformList;
+}
+
+const GlslValidator::ProgramInputMap& GlslValidator::updateAttributesList()
+{
+    ShaderValidationErrorList errors;
+    const std::string errorType("GLSL attribute parsing error.");
+
+    if (_attributeList.size() > 0)
+    {
+        return _attributeList;
+    }
+
+    if (_programId <= 0)
+    {
+        errors.push_back("Cannot parse for attributes without a valid program");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+
+    GLint numAttributes = 0;
+    GLint maxNameLength = 0;
+    glGetProgramiv(_programId, GL_ACTIVE_ATTRIBUTES, &numAttributes);
+    glGetProgramiv(_programId, GL_ACTIVE_ATTRIBUTE_MAX_LENGTH, &maxNameLength);
+    char* attributeName = new char[maxNameLength];
+
+    for (int i = 0; i < numAttributes; i++)
+    {
+        GLint attributeSize = 0;
+        GLenum attributeType = 0;
+        glGetActiveAttrib(_programId, GLuint(i), maxNameLength, nullptr, &attributeSize, &attributeType, attributeName);
+        GLint attributeLocation = glGetAttribLocation(_programId, attributeName);
+        if (attributeLocation >= 0)
+        {
+            ProgramInputPtr inputPtr = std::make_shared<ProgramInput>(attributeLocation, attributeType, attributeSize);
+            _attributeList[std::string(attributeName)] = inputPtr;
+            //std::cout << "Scanned attribute : " << attributeName << ". Location: " << attributeLocation << ". Type: " << attributeType << std::endl;
+        }
+    }
+    delete[] attributeName;
+
+    return _attributeList;
+}
+
+void GlslValidator::bindTimeAndFrame()
+{
+    ShaderValidationErrorList errors;
+
+    if (_programId <= 0)
+    {
+        const std::string errorType("GLSL input binding error.");
+        errors.push_back("Cannot bind time/frame without a valid program");
+        throw ExceptionShaderValidationError(errorType, errors);
+    }
+
+    GLint location = UNDEFINED_OPENGL_PROGRAM_LOCATION;
+
+    // Bind time
+    auto programInput = _uniformList.find("u_time");
+    if (programInput != _uniformList.end())
+    {
+        location = programInput->second->_location;
+        if (location >= 0)
+        {
+            //std::cout << "Bind time. Location: " << location << std::endl;
+            glUniform1f(location, 1.0f);
+        }
+    }
+
+    // Bind frame
+    programInput = _uniformList.find("u_frame");
+    if (programInput != _uniformList.end())
+    {
+        location = programInput->second->_location;
+        if (location >= 0)
+        {
+            //std::cout << "Bind frame. Location: " << location << std::endl;
+            glUniform1f(location, 1.0f);
+        }
+    }
+}
+
+
+void GlslValidator::bindViewInformation()
```

Comment:
Bind view information. Still needs a bit of work for some matrices. I think we may want to reduce this a bit but basic modeview and projection are being passed through currently.